src/main/java/bio/terra/cli/service/WorkspaceManagerService.java

```diff
@@ -656,7 +658,9 @@ public GcpBigQueryDatasetResource createControlledBigQueryDataset(
             .dataset(
                 new GcpBigQueryDatasetCreationParameters()
                     .datasetId(createParams.datasetId)
-                    .location(createParams.location));
+                    .location(createParams.location)
+                    .defaultPartitionLifetime(createParams.defaultPartitionLifetime)
```

Comment:
this is coupling our userfacing serialization very directly to the WSM API. We need their semantics around 0 and nulls to be identical, which makes me a little nervous.

Reply:
Would your concern be addressed by not calling the respective setter on the (WSM client library defined) `GcpBigQueryDatasetCreationParameters` if the corresponding parameter is `null`?

Reply:
Sorry, I'm not following. If the corresponding parameter, say `x` is null, what's the difference between `wsmParams.x(createParams.x)` and `if createParams.x != null { wsmParams.x.(createParams.x); }`
I'm mostly nervous about us changing the semantics of edge case fields in the userfacing serialization or the WSM api and us not remembering to change this code. But introducing a translation layer is maybe overkill. But magic 0 numbers & Integer null semantics make me a little nervous.

Reply:
I agree that coupling user-facing serialization with the WSM API serialization seems like a bad idea. But I think that's addressed here by having separate classes (`CreateBqDatasetParams` and `GcpBigQueryDatasetCreationParameters`) and an explicit translation between them. Perhaps a comment here would be helpful that both serialization classes (and BQ) have the same semantics around special zeros and nulls, and that's why the translation is pretty trivial? Currently we have examples of both trivial translations (e.g. for [AI notebook creation](https://github.com/DataBiosphere/terra-cli/blob/60bb9b54659a3f0376262f6a58a7b200d5b62860/src/main/java/bio/terra/cli/service/WorkspaceManagerService.java#L521)) and more involved ones (e.g. for [bucket lifecycle rules](https://github.com/DataBiosphere/terra-cli/blob/60bb9b54659a3f0376262f6a58a7b200d5b62860/src/main/java/bio/terra/cli/service/WorkspaceManagerService.java#L597)).

Reply:
OK, I think I get it; the WSM side could offer an API that explicitly says “clear the expiration”, in which case we’d want to map (user input of) zero to “clear”, null to “don’t touch it”, and non-zero to “set”.  Your concern is that we make this change in WSM and it doesn’t propagate here?  I understand the concern but will leave this for the moment as I can’t think of a change that adds value on this front.